src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+                PreparedStatement statement = dbConnection
+                        .prepareStatement("SELECT * FROM manufacturers"
+                                + " WHERE is_deleted = 0 AND id = ?;")) {
+            statement.setString(1, String.valueOf(id.toString().charAt(0)));
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return manufacturerBuilder(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> manufacturerList = new ArrayList<>();
+        try (Connection dbConnection = ConnectionUtil.getConnection();
+                PreparedStatement statement = dbConnection
+                        .prepareStatement("SELECT * FROM manufacturers WHERE is_deleted = 0;")) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Optional<Manufacturer> manufacturerOptional = manufacturerBuilder(resultSet);
+                if (manufacturerOptional.isEmpty()) {
+                    continue;
+                }
+                manufacturerList.add(manufacturerOptional.get());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get list of manufacturers", e);
+        }
+        return manufacturerList;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        try (Connection dbConnection = ConnectionUtil.getConnection();
+                PreparedStatement statement = dbConnection
+                        .prepareStatement("UPDATE manufacturers SET name = ?,"
+                        + " country = ? WHERE id = ? AND is_deleted = 0")) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.setLong(3, manufacturer.getId());
+            statement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer: "
+                    + manufacturer.toString(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection dbConnection = ConnectionUtil.getConnection();
+                PreparedStatement statement = dbConnection.prepareStatement("UPDATE manufacturers "
+                        + "SET is_deleted = true WHERE id = ?")) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer at id " + id, e);
+        }
+    }
+
+    private Optional<Manufacturer> manufacturerBuilder(ResultSet resultSet) {
```

Comment:
```suggestion
    private Optional<Manufacturer> getManufacturer(ResultSet resultSet) throws SQLException {
```
we can just throw this exception in method signature because we are catching it in dao methods anyway